benchmark/benchmarks.zig

```diff
@@ -0,0 +1,116 @@
+const std = @import("std");
+const testing = std.testing;
+const protobuf = @import("protobuf");
+const metrics = @import("./generated/opentelemetry/proto/metrics/v1.pb.zig");
+const benchmark_data = @import("./generated/benchmark.pb.zig");
+const zbench = @import("zbench");
+const dataset_generator = @import("./generate_dataset.zig");
+
+const DATASET_FILENAME = "test.data";
+
+fn bench_encode(allocator: std.mem.Allocator) void {
+    var buffer: [1]u8 = .{0};
+    var w: std.Io.Writer.Discarding = .init(&buffer);
+    _ = input_to_encode.encode(&w.writer, allocator) catch @panic("Failed to encode");
+}
+
+fn bench_encode_arena(allocator: std.mem.Allocator) void {
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    defer arena.deinit();
+    const arena_allocator = arena.allocator();
+    var buffer: [1]u8 = .{0};
+    var w: std.Io.Writer.Discarding = .init(&buffer);
+    _ = input_to_encode.encode(&w.writer, arena_allocator) catch @panic("Failed to encode");
+}
+
+fn bench_decode(allocator: std.mem.Allocator) void {
+    var reader: std.Io.Reader = .fixed(input_to_decode);
+    _ = benchmark_data.BenchmarkData.decode(&reader, allocator) catch @panic("Failed to decode");
+}
+
+const DataSet = struct {
+    data: benchmark_data.BenchmarkData,
+    encoded: []u8,
+};
+
+fn loadFixedDataset(allocator: std.mem.Allocator) !DataSet {
+    // Try to open the test.data file
+    std.debug.print("Loading dataset from {s}...\n", .{DATASET_FILENAME});
+
+    const file = try std.fs.cwd().openFile(DATASET_FILENAME, .{});
```

Comment:
nit: clarification for future contributors

```suggestion
    // NOTE: it is important to always load the file from disk at runtime to prevent
    // early optimizations of the compiler that may impact(tint) the benchmark
    const file = try std.fs.cwd().openFile(DATASET_FILENAME, .{});
```